tests/test_algorithm.py

```diff
@@ -726,9 +726,37 @@ def test_future_symbol(self):
         with self.assertRaises(TypeError):
             algo.future_symbol({'foo': 'bar'})
 
+    def test_future_chain_offset(self):
+        algo = TradingAlgorithm(env=self.env)
+        algo.datetime = pd.Timestamp('2006-12-01', tz='UTC')
+
+        self.assertEqual(
+            algo.future_chain('CL', offset=1).as_of_date,
+            self.trading_calendar.next_session_label(
+                algo.datetime
+            )
+        )
+
+        self.assertEqual(
+            algo.future_chain("CL", offset=5).as_of_date,
+            pd.Timestamp("2006-12-08", tz='UTC')
```

Comment:
I'm fine with doing that for all of them.  Might be helpful to add the calendar for the reader:

```
In [24]: calendar.prmonth(2006, 12)
   December 2006
Mo Tu We Th Fr Sa Su
             1  2  3
 4  5  6  7  8  9 10
11 12 13 14 15 16 17
18 19 20 21 22 23 24
25 26 27 28 29 30 31
```